examples/math/smooth_follow.rs

```diff
@@ -0,0 +1,145 @@
+#[derive(Resource)]
+struct TargetPosition(Vec3);
+
+// The decay rate used by the smooth following:
+#[derive(Resource)]
+struct DecayRate(f32);
+
+// The sphere that follows the target sphere by moving towards it with nudging:
+#[derive(Component)]
+struct FollowingSphere;
+
+/// The source of randomness used by this example.
+#[derive(Resource)]
+struct RandomSource(ChaCha8Rng);
+
+fn setup(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    // A plane:
+    commands.spawn(PbrBundle {
+        mesh: meshes.add(Plane3d::default().mesh().size(12.0, 12.0)),
+        material: materials.add(Color::srgb(0.3, 0.15, 0.3)),
+        transform: Transform::from_xyz(0.0, -2.5, 0.0),
+        ..default()
+    });
+
+    // The target sphere:
+    commands.spawn((
+        PbrBundle {
+            mesh: meshes.add(Sphere::new(0.3)),
+            material: materials.add(Color::srgb(0.3, 0.15, 0.9)),
+            ..default()
+        },
+        TargetSphere,
+    ));
+
+    // The sphere that follows it:
+    commands.spawn((
+        PbrBundle {
+            mesh: meshes.add(Sphere::new(0.3)),
+            material: materials.add(Color::srgb(0.9, 0.3, 0.3)),
+            transform: Transform::from_translation(vec3(0.0, -2.0, 0.0)),
+            ..default()
+        },
+        FollowingSphere,
+    ));
+
+    // A light:
+    commands.spawn(PointLightBundle {
+        point_light: PointLight {
+            intensity: 15_000_000.0,
+            shadows_enabled: true,
+            ..default()
+        },
+        transform: Transform::from_xyz(4.0, 8.0, 4.0),
+        ..default()
+    });
+
+    // A camera:
+    commands.spawn(Camera3dBundle {
+        transform: Transform::from_xyz(-2.0, 3.0, 5.0).looking_at(Vec3::ZERO, Vec3::Y),
+        ..default()
+    });
+
+    // Set starting values for resources used by the systems:
+    commands.insert_resource(TargetSphereSpeed(5.0));
+    commands.insert_resource(DecayRate(2.0));
+    commands.insert_resource(TargetPosition(Vec3::ZERO));
+    commands.insert_resource(RandomSource(ChaCha8Rng::seed_from_u64(68941654987813521)));
+}
+
+fn move_target(
+    mut target: Query<&mut Transform, With<TargetSphere>>,
+    target_speed: Res<TargetSphereSpeed>,
+    mut target_pos: ResMut<TargetPosition>,
+    time: Res<Time>,
+    mut rng: ResMut<RandomSource>,
+) {
+    let mut target = target.single_mut();
+
+    match Dir3::new(target_pos.0 - target.translation) {
+        // The target and the present position of the target sphere are far enough to have a well-
+        // defined direction between them, so let's move closer:
+        Ok(dir) => {
+            let delta_time = time.delta_seconds();
+            let abs_delta = (target_pos.0 - target.translation).norm();
+
+            // Avoid overshooting in case of high values of `delta_time``:
+            let magnitude = min_by(abs_delta, delta_time * target_speed.0, |f0, f1| {
```

Comment:
Wouldn't

```rust
let magnitude = abs_delta.min(delta_time * target_speed.0);
```

work here?

Reply:
I forgot about `f32::min` and was just thinking of the `Ord` version. Good catch!